File: models/Gpt-3.5-turbo/cpp/code/problems/introductory_problems/Tower_of_Hanoi.cpp

#include <iostream>
#include <vector>

using namespace std;

vector<pair<int, int>> moves; // to store the moves

void towerOfHanoi(int n, int source, int target, int auxiliary) {
    if (n > 0) {
        // Move n-1 disks from source to auxiliary
        towerOfHanoi(n - 1, source, auxiliary, target);
        
        // Move the nth disk from source to target
        moves.push_back({source, target});

        // Move the n-1 disks from auxiliary to target
        towerOfHanoi(n - 1, auxiliary, target, source);
    }
}

int main() {
    int n;
    cin >> n;
    
    towerOfHanoi(n, 1, 3, 2); // start from source stack 1, target stack 3, and auxiliary stack 2
    
    cout << moves.size() << endl;
    for (const auto& move : moves) {
        cout << move.first << " " << move.second << endl;
    }
    
    return 0;
}

File: models/Gpt-3.5-turbo/py/code/introductory_problems/Tower_of_Hanoi.py

def tower_of_hanoi(n, source, auxiliary, destination):
    if n == 1:
        print(source, destination)
        return 1
    else:
        moves = 0
        moves += tower_of_hanoi(n-1, source, destination, auxiliary)
        print(source, destination)
        moves += 1
        moves += tower_of_hanoi(n-1, auxiliary, source, destination)
        return moves

n = int(input())
print(tower_of_hanoi(n, 1, 2, 3))
